nucypher/datastore/queries.py

```diff
@@ -0,0 +1,56 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+from typing import List, Type
+
+from nucypher.datastore.base import DatastoreRecord
+from nucypher.datastore.datastore import Datastore, RecordNotFound
+from nucypher.datastore.models import PolicyArrangement, TreasureMap, Workorder
+
+
+def find_expired_policies(ds: Datastore, now) -> List[Type['DatastoreRecord']]:
+    try:
+        return ds.query_by(PolicyArrangement,
+                           filter_field='expiration',
+                           filter_func=lambda expiration: expiration <= now,
+                           writeable=True)
+    except RecordNotFound:
+        return []
```

Comment:
Ah yes, I like this pattern a lot, it makes the calling code much clearer.

Reply:
Unfortunately, this pattern doesn't work and this actually is the reason for failing tests - the evaluation of query is lazy so the `RecordNotFound` will be raised further down the road in a different scope.

I'm going to fix this.

Reply:
I committed an alternative query pattern. It's not as ergonomic, but it contains a necessary workaround for writable records: a lot of times, we want to `find_*` records and then `record.delete()` them.